src/ui/controllercollection.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Manages UI Controllers.
+ *
+ * @class ControllerCollection
+ * @extends Collection
+ */
+CKEDITOR.define( [
+	'collection',
+	'ckeditorerror'
+], ( Collection, CKEditorError ) => {
+	class ControllerCollection extends Collection {
```

Comment:
Is this class really necessary? Maybe we can add `opts.name` to the `Collection` class. Would make less code.

The counterargument is that _perhaps_ this class will get more behaviour in the future. But as we don't have any now, maybe let's KISS.